### PR TITLE
ale 0.8.6->0.8.8

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by amystamile-usgs** [comment]
Test Failures:

	990 - lro_module_test_lroc (Failed)
	1032 - mex_module_test_hrsc (Failed)
	1435 - DefaultCube.FunctionalTestCkwriterDefault (Failed)
	1436 - DefaultCube.FunctionalTestCkwriterFromlist (Failed)

**Comment by amystamile-usgs** [comment]
@acpaquette Alot of new issues with the lroc module test. Seems like there are issues with the Sun Information with the ale update and alot of changes with the naif keywords in the label. Also seeing differences in the number of records for the tables. Any ideas what could be causing this?

Screen Shot 2023-05-06 at 8 10 05 AM



**Truth UV PVL NaifKeywords**
```
Object = NaifKeywords
  BODY_CODE                                 = 301
  BODY301_RADII                             = (1737.4, 1737.4, 1737.4)
  BODY_FRAME_CODE                           = 31001
  INS-85626_SWAP_OBSERVER_TARGET            = TRUE
  INS-85626_LIGHTTIME_CORRECTION            = NONE
  INS-85626_LT_SURFACE_CORRECT              = TRUE
  INS-85626_FOCAL_LENGTH                    = 4.693
  INS-85626_PIXEL_PITCH                     = 0.009
  CLOCK_ET_-85_1/274692469:15073_COMPUTED   = 8d506af75541b241
  INS-85626_FILTER_BANDCENTER_SIZE_COMPUTED = 2
  INS-85626_FILTER_BANDCENTER               = (321, 360)
  INS-85626_FILTER_OFFSET_SIZE_COMPUTED     = 2
  INS-85626_FILTER_OFFSET                   = (244, 302)
  INS-85626_FILTER_BANDID_SIZE_COMPUTED     = 2
  INS-85626_FILTER_BANDID                   = (1, 2)
  INS-85641_FOCAL_LENGTH                    = 4.6690884
  INS-85641_BORESIGHT_SAMPLE                = 513.5482
  INS-85641_BORESIGHT_LINE                  = 295.4784
  INS-85642_FOCAL_LENGTH                    = 4.7120818
  INS-85642_BORESIGHT_SAMPLE                = 513.5601
  INS-85642_BORESIGHT_LINE                  = 295.4764
  INS-85626_COLOR_SAMPLE_OFFSET             = 264
  INS-85626_TRANSX                          = (0.0, 0.0, -0.009)
  INS-85626_TRANSY                          = (0.0, 0.009, 0.0)
  INS-85626_ITRANSS                         = (0.0, 0.0, 111.11111111111)
  INS-85626_ITRANSL                         = (0.0, -111.11111111111, 0.0)
  INS-85641_TRANSX                          = (0.0, 3.11994535e-07,
                                               -0.008999999994592)
  INS-85641_TRANSY                          = (0.0, 0.008999999994592,
                                               3.11994535e-07)
  INS-85641_ITRANSS                         = (0.0, -0.0038517843825,
                                               111.11111104435)
  INS-85641_ITRANSL                         = (0.0, -111.11111104435,
                                               -0.0038517843825)
  INS-85641_OD_K                            = (0.025824582751555,
                                               4.66139483592234e-05,
                                               1.44650735580122e-04)
  INS-85642_TRANSX                          = (0.0, 1.425859353e-06,
                                               -0.008999999887051)
  INS-85642_TRANSY                          = (0.0, 0.008999999887051,
                                               1.425859353e-06)
  INS-85642_ITRANSS                         = (0.0, 0.017603201893,
                                               111.11110971668)
  INS-85642_ITRANSL                         = (0.0, -111.11110971668,
                                               0.017603201893)
  INS-85642_OD_K                            = (0.025144400734802,
                                               2.2601061670964e-04,
                                               1.24071862326273e-04)
End_Object
```

**Output UV PVL NaifKeywords**
```
Object = NaifKeywords
  BODY301_LONG_AXIS                         = 0
  BODY301_NUT_PREC_DEC                      = (1.5419, 0.0239, -0.0278,
                                               0.0068, 0, -0.0029, 0.0009, 0,
                                               0, 0.0008)
  BODY301_NUT_PREC_PM                       = (3.561, 0.1208,
                                               -0.06419999999999999, 0.0158,
                                               0.0252, -0.0066, -0.0047,
                                               -0.0046, 0.0028, 0.0052)
  BODY301_NUT_PREC_RA                       = (-3.8787, -0.1204,
                                               0.07000000000000001, -0.0172, 0,
                                               0.0072, 0, 0, 0, -0.0052)
  BODY301_PM                                = (38.3213, 13.17635815, -1.4e-12)
  BODY301_POLE_DEC                          = (66.53919999999999, 0.013, 0)
  BODY301_POLE_RA                           = (269.9949, 0.0031, 0)
  BODY301_RADII                             = (1737.4, 1737.4, 1737.4)
  BODY_CODE                                 = 301
  BODY_FRAME_CODE                           = 31001
  FRAME_-85626_CENTER                       = -85
  FRAME_-85626_CLASS                        = 4
  FRAME_-85626_CLASS_ID                     = -85626
  FRAME_-85626_NAME                         = LRO_LROCWAC_UV
  FRAME_-85641_CENTER                       = -85
  FRAME_-85641_CLASS                        = 4
  FRAME_-85641_CLASS_ID                     = -85641
  FRAME_-85641_NAME                         = LRO_LROCWAC_UV_FILTER_1
  INS-85600_BORESIGHT_LINE                  = 0
  INS-85600_BORESIGHT_SAMPLE                = 2548
  INS-85600_FOCAL_LENGTH                    = 699.62
  INS-85600_ITRANSL                         = (0, 142.857, 0)
  INS-85600_ITRANSS                         = (0, 0, 142.857)
  INS-85600_OD_K                            = 1.81e-05
  INS-85600_TRANSX                          = (0, 0, 0.007)
  INS-85600_TRANSY                          = (0, 0.007, 0)
  INS-85610_BORESIGHT_LINE                  = 0
  INS-85610_BORESIGHT_SAMPLE                = 2496
  INS-85610_FOCAL_LENGTH                    = 701.5700000000001
  INS-85610_ITRANSL                         = (0, -142.8571, 0)
  INS-85610_ITRANSS                         = (0, 0, -142.8571)
  INS-85610_OD_K                            = 1.83e-05
  INS-85610_TRANSX                          = (0, 0, -0.007)
  INS-85610_TRANSY                          = (0, -0.007, 0)
  INS-85621_BORESIGHT_LINE                  = 778.2
  INS-85621_BORESIGHT_SAMPLE                = 513.9
  INS-85621_FOCAL_LENGTH                    = 6.013
  INS-85621_ITRANSL                         = (0, -111.11111111111, 0)
  INS-85621_ITRANSS                         = (0, 0, 111.11111111111)
  INS-85621_OD_K                            = (-0.009900000000000001, -0.0005)
  INS-85621_TRANSX                          = (0, 0, -0.008999999999999999)
  INS-85621_TRANSY                          = (0, 0.008999999999999999, 0)
  INS-85626_BORESIGHT                       = (0, 0, 1)
  INS-85626_BORESIGHT_LINE                  = 298.1
  INS-85626_BORESIGHT_SAMPLE                = 515.3
  INS-85626_BW0_SAMPLE_OFFSET               = 264
  INS-85626_BW1_SAMPLE_OFFSET               = 264
  INS-85626_BW2_SAMPLE_OFFSET               = 0
  INS-85626_BW3_SAMPLE_OFFSET               = 0
  INS-85626_CCD_CENTER                      = (511.5, 511.5)
  INS-85626_CK_FRAME_ID                     = -85000
  INS-85626_CK_REFERENCE_ID                 = 1
  INS-85626_COLOR_SAMPLE_OFFSET             = 264
  INS-85626_F/RATIO                         = 5.65
  INS-85626_FILTER_BANDCENTER               = (321, 360)
  INS-85626_FILTER_BANDID                   = (1, 2)
  INS-85626_FILTER_BANDWIDTH                = (32, 15)
  INS-85626_FILTER_OFFSET                   = (244, 302)
  INS-85626_FOCAL_LENGTH                    = 4.693
  INS-85626_FOV_ANGLE_UNITS                 = DEGREES
  INS-85626_FOV_ANGULAR_SIZE                = (1.05, 0.0307)
  INS-85626_FOV_CLASS_SPEC                  = ANGLES
  INS-85626_FOV_CROSS_ANGLE                 = 30.1
  INS-85626_FOV_FRAME                       = LRO_LROCWAC_UV
  INS-85626_FOV_REF_ANGLE                   = 30.1
  INS-85626_FOV_REF_VECTOR                  = (0, 1, 0)
  INS-85626_FOV_SHAPE                       = RECTANGLE
  INS-85626_IFOV                            = 0.00198
  INS-85626_ITRANSL                         = (0, -111.11111111111, 0)
  INS-85626_ITRANSS                         = (0, 0, 111.11111111111)
  INS-85626_LIGHTTIME_CORRECTION            = NONE
  INS-85626_LT_SURFACE_CORRECT              = TRUE
  INS-85626_OD_K                            = (-0.024, -0.0007)
  INS-85626_PIXEL_LINES                     = 1024
  INS-85626_PIXEL_PITCH                     = 0.008999999999999999
  INS-85626_PIXEL_SAMPLES                   = 1024
  INS-85626_PIXEL_SIZE                      = (9, 9)
  INS-85626_PLATFORM_ID                     = -85620
  INS-85626_SWAP_OBSERVER_TARGET            = TRUE
  INS-85626_TRANSX                          = (0, 0, -0.008999999999999999)
  INS-85626_TRANSY                          = (0, 0.008999999999999999, 0)
  INS-85626_UV_SAMPLE_OFFSET                = 264
  INS-85631_BORESIGHT_LINE                  = 775.8181
  INS-85631_BORESIGHT_SAMPLE                = 509.5696
  INS-85631_FOCAL_LENGTH                    = 5.9983909
  INS-85631_ITRANSL                         = (0, -111.1111108720638,
                                               0.0072884596481)
  INS-85631_ITRANSS                         = (0, 0.0072884596481,
                                               111.1111108720638)
  INS-85631_OD_K                            = (0.0113867116753854,
                                               0.00014704581347987,
                                               5.038005020438671e-06)
  INS-85631_TRANSX                          = (0, 5.90365231e-07,
                                               -0.008999999980637)
  INS-85631_TRANSY                          = (0, 0.008999999980637,
                                               5.90365231e-07)
  INS-85632_BORESIGHT_LINE                  = 775.7541
  INS-85632_BORESIGHT_SAMPLE                = 509.5445
  INS-85632_FOCAL_LENGTH                    = 6.0193387
  INS-85632_ITRANSL                         = (0, -111.1111110790686,
                                               0.0026684381921)
  INS-85632_ITRANSS                         = (0, 0.0026684381921,
                                               111.1111110790686)
  INS-85632_OD_K                            = (0.0113301372841102,
                                               0.000147788496863675,
                                               4.87090732732635e-06)
  INS-85632_TRANSX                          = (0, 2.16143494e-07,
                                               -0.008999999997405)
  INS-85632_TRANSY                          = (0, 0.008999999997405,
                                               2.16143494e-07)
  INS-85633_BORESIGHT_LINE                  = 775.7578
  INS-85633_BORESIGHT_SAMPLE                = 509.5379
  INS-85633_FOCAL_LENGTH                    = 6.0247535
  INS-85633_ITRANSL                         = (0, -111.1111111060689,
                                               -0.0010585309475)
  INS-85633_ITRANSS                         = (0, -0.0010585309475,
                                               111.1111111060689)
  INS-85633_OD_K                            = (0.0113029012587659,
                                               0.000147434788741622,
                                               4.92841571946775e-06)
  INS-85633_TRANSX                          = (0, -8.5741007e-08,
                                               -0.008999999999591999)
  INS-85633_TRANSY                          = (0, 0.008999999999591999,
                                               -8.5741007e-08)
  INS-85634_BORESIGHT_LINE                  = 775.7545
  INS-85634_BORESIGHT_SAMPLE                = 509.5297
  INS-85634_FOCAL_LENGTH                    = 6.0308669
  INS-85634_ITRANSL                         = (0, -111.1111064716567,
                                               -0.0321090306666)
  INS-85634_ITRANSS                         = (0, -0.0321090306666,
                                               111.1111064716567)
  INS-85634_OD_K                            = (0.0113109452166359,
                                               0.000144463288593614,
                                               4.88754251291127e-06)
  INS-85634_TRANSX                          = (0, -2.600831484e-06,
                                               -0.008999999624204001)
  INS-85634_TRANSY                          = (0, 0.008999999624204001,
                                               -2.600831484e-06)
  INS-85635_BORESIGHT_LINE                  = 775.7604
  INS-85635_BORESIGHT_SAMPLE                = 509.5314
  INS-85635_FOCAL_LENGTH                    = 6.0377111
  INS-85635_ITRANSL                         = (0, -111.1111108046414,
                                               -0.008252538114899999)
  INS-85635_ITRANSS                         = (0, -0.008252538114899999,
                                               111.1111108046414)
  INS-85635_OD_K                            = (0.0112564226546747,
                                               0.000146609439724495,
                                               4.80869164345756e-06)
  INS-85635_TRANSX                          = (0, -6.68455587e-07,
                                               -0.008999999975176001)
  INS-85635_TRANSY                          = (0, 0.008999999975176001,
                                               -6.68455587e-07)
  INS-85641_BORESIGHT                       = (0, 0, 1)
  INS-85641_BORESIGHT_LINE                  = 295.4784
  INS-85641_BORESIGHT_SAMPLE                = 513.5482
  INS-85641_BW0_SAMPLE_OFFSET               = 264
  INS-85641_BW1_SAMPLE_OFFSET               = 264
  INS-85641_BW2_SAMPLE_OFFSET               = 0
  INS-85641_BW3_SAMPLE_OFFSET               = 0
  INS-85641_CCD_CENTER                      = (512.5, 512.5)
  INS-85641_CK_FRAME_ID                     = -85000
  INS-85641_CK_REFERENCE_ID                 = 1
  INS-85641_COLOR_SAMPLE_OFFSET             = 264
  INS-85641_F/RATIO                         = 5.65
  INS-85641_FILTER_BANDCENTER               = 321
  INS-85641_FILTER_BANDID                   = 1
  INS-85641_FILTER_BANDWIDTH                = 32
  INS-85641_FILTER_OFFSET                   = 244
  INS-85641_FOCAL_LENGTH                    = 4.6690884
  INS-85641_FOV_BOUNDARY_CORNERS            = (0.1113, -0.55241, 1, 0.1113,
                                               -0.55241, 1,
                                               0.09684100000000001, 0, 1,
                                               0.11189)
  INS-85641_FOV_FRAME                       = LRO_LROCWAC_UV_FILTER_1
  INS-85641_FOV_SHAPE                       = POLYGON
  INS-85641_IFOV                            = 0.0019276
  INS-85641_ITRANSL                         = (0, -111.111111044348,
                                               -0.0038517843825)
  INS-85641_ITRANSS                         = (0, -0.0038517843825,
                                               111.111111044348)
  INS-85641_LIGHTTIME_CORRECTION            = NONE
  INS-85641_LT_SURFACE_CORRECT              = TRUE
  INS-85641_OD_K                            = (0.0258245827515546,
                                               4.66139483592234e-05,
                                               0.000144650735580122)
  INS-85641_PIXEL_LINES                     = 1024
  INS-85641_PIXEL_PITCH                     = 0.008999999999999999
  INS-85641_PIXEL_SAMPLES                   = 1024
  INS-85641_PIXEL_SIZE                      = (0.008999999999999999,
                                               0.008999999999999999)
  INS-85641_PLATFORM_ID                     = -85620
  INS-85641_SWAP_OBSERVER_TARGET            = TRUE
  INS-85641_TRANSX                          = (0, 3.11994535e-07,
                                               -0.008999999994592001)
  INS-85641_TRANSY                          = (0, 0.008999999994592001,
                                               3.11994535e-07)
  INS-85641_UV_SAMPLE_OFFSET                = 264
  INS-85642_BORESIGHT_LINE                  = 295.4764
  INS-85642_BORESIGHT_SAMPLE                = 513.5601
  INS-85642_FOCAL_LENGTH                    = 4.7120818
  INS-85642_ITRANSL                         = (0, -111.1111097166839,
                                               0.017603201893)
  INS-85642_ITRANSS                         = (0, 0.017603201893,
                                               111.1111097166839)
  INS-85642_OD_K                            = (0.025144400734802,
                                               0.00022601061670964,
                                               0.000124071862326273)
  INS-85642_TRANSX                          = (0, 1.425859353e-06,
                                               -0.008999999887051)
  INS-85642_TRANSY                          = (0, 0.008999999887051,
                                               1.425859353e-06)
  TKFRAME_-85626_ANGLES                     = (-0.140354, -1.14619,
                                               -0.09951699999999999)
  TKFRAME_-85626_AXES                       = (1, 2, 3)
  TKFRAME_-85626_RELATIVE                   = LRO_LROCWAC
  TKFRAME_-85626_SPEC                       = ANGLES
  TKFRAME_-85626_UNITS                      = DEGREES
  TKFRAME_-85641_ANGLES                     = (0, 0, 0)
  TKFRAME_-85641_AXES                       = (1, 2, 3)
  TKFRAME_-85641_RELATIVE                   = LRO_LROCWAC_UV
  TKFRAME_-85641_SPEC                       = ANGLES
  TKFRAME_-85641_UNITS                      = DEGREES
  CLOCK_ET_-85_1/274692469:15073_COMPUTED   = 8d506af75541b241
  INS-85626_FILTER_BANDCENTER_SIZE_COMPUTED = 0
  INS-85626_FILTER_OFFSET_SIZE_COMPUTED     = 0
  INS-85626_FILTER_BANDID_SIZE_COMPUTED     = 0
End_Object
```

File: isis/tests/FunctionalTestsCkwriter.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ TEST_F(DefaultCube, FunctionalTestCkwriterDefault) {
   newKernelRotation->SetEphemerisTime(startTime);
   originalRotation->SetEphemerisTime(startTime);
 
-  ASSERT_EQ(newKernelRotation->cacheSize(), originalRotation->cacheSize());
+  ASSERT_EQ(newKernelRotation->cacheSize(), 2);
```
Comment (amystamile-usgs) [inline isis/tests/FunctionalTestsCkwriter.cpp:70]: @acpaquette also seeing the cacheSize going from 1 to 2.

**Comment by acpaquette** [comment]
Just so there is some known action on this that Amy and I talked about. The pushframe times in ALE differ from ISIS which results in some difference between the ALE spiceinit'd images and the ISIS spiceinit'd images.

Second, ALE doesn't really handle kernel order well. In ISIS, there is an expectation that certain kernels will load AFTER others. In ISIS this is the order of kernels in each field, followed by any extra kernels passed by the user. ALE does a super basic sort of the kernels with some ordering applied to certain kernel types. This is seems to be the error we are seeing from some of our users who are passing their own kernels through the extras parameter.

**Comment by AustinSanders** [comment]
@acpaquette Are these concerns the results of this PR or are they existing in ALE?  Can this get pushed through?

**Comment by acpaquette** [comment]
I am not sure in this case. The change in the cacheSize seems strange as I don't think(?) ALE should be changing it. I am working on this locally and @amystamile-usgs may have more insight the cacheSize problem

**Comment by acpaquette** [comment]
So the cache size issue was fixed here:
https://github.com/DOI-USGS/ale/pull/521/files#diff-6bb21563c0d335c00665df46999166f45ba78c2c2e39abfa7418e3ac42cd69d2

This fix is not present in 0.8.8, if we get 0.9.0 out this problem will be fixed. The only reason the problem started was because I fixed the viking driver to be picked up by ISIS. Fun little rabbit hole

**Comment by amystamile-usgs** [comment]
> So the cache size issue was fixed here: https://github.com/DOI-USGS/ale/pull/521/files#diff-6bb21563c0d335c00665df46999166f45ba78c2c2e39abfa7418e3ac42cd69d2
> 
> This fix is not present in 0.8.8, if we get 0.9.0 out this problem will be fixed. The only reason the problem started was because I fixed the viking driver to be picked up by ISIS. Fun little rabbit hole

@acpaquette I am glad you figured this out. I would have not been able to find this. Should we just scrap this PR and update to 0.9.0 instead once it's released?

**Comment by amystamile-usgs** [comment]
Skipping 0.8.8 to 0.9.0

**Pull request closed without merging**